src/finch-mcp-server/awslabs/finch_mcp_server/utils/vm.py

```diff
@@ -0,0 +1,397 @@
+
+
+def get_vm_status() -> subprocess.CompletedProcess:
+    """Get the current status of the Finch VM.
+
+    This function executes 'finch vm status' and returns the raw result.
+    It's a wrapper around execute_command that simplifies checking
+    the VM status, which is a common operation used by multiple tools.
+
+    Returns:
+        CompletedProcess with the status command result, containing:
+        - returncode: The exit code of the command
+        - stdout: Standard output containing status information
+        - stderr: Standard error output, may also contain status information
+
+    """
+    return execute_command(['finch', 'vm', 'status'])
+
+
+def is_vm_nonexistent(status_result: subprocess.CompletedProcess) -> bool:
+    """Check if the Finch VM is nonexistent based on status result.
+
+    This function analyzes the output of 'finch vm status' to determine
+    if the VM has not been created yet.
+
+    Args:
+        status_result: CompletedProcess object from running 'finch vm status'
+
+    Returns:
+        bool: True if the VM is nonexistent, False otherwise
+
+    """
+    return (
+        'nonexistent' in status_result.stderr.lower()
+        or 'nonexistent' in status_result.stdout.lower()
+    )
+
+
+def is_vm_stopped(status_result: subprocess.CompletedProcess) -> bool:
+    """Check if the Finch VM is stopped based on status result.
+
+    This function analyzes the output of 'finch vm status' to determine
+    if the VM exists but is not currently running.
+
+    Args:
+        status_result: CompletedProcess object from running 'finch vm status'
+
+    Returns:
+        bool: True if the VM is stopped, False otherwise
+
+    """
+    return 'stopped' in status_result.stderr.lower() or 'stopped' in status_result.stdout.lower()
+
+
+def is_vm_running(status_result: subprocess.CompletedProcess) -> bool:
+    """Check if the Finch VM is running based on status result.
+
+    This function analyzes the output of 'finch vm status' to determine
+    if the VM is currently active and operational.
+
+    Args:
+        status_result: CompletedProcess object from running 'finch vm status'
+
+    Returns:
+        bool: True if the VM is running, False otherwise
+
+    """
+    return 'running' in status_result.stdout.lower() or 'running' in status_result.stderr.lower()
+
+
+def initialize_vm() -> Dict[str, str]:
+    """Initialize a new Finch VM.
+
+    This function runs 'finch vm init' to create a new Finch VM instance.
+    It's used when the VM doesn't exist yet and needs to be created.
+
+    Returns:
+        Dict[str, Any]: Result dictionary with:
+            - status: "success" if initialization succeeded, "error" otherwise
+            - message: Details about the initialization result
+
+    """
+    if sys.platform == 'linux':
+        logger.debug('Linux OS detected. No VM operation required')
+        return format_result(STATUS_SUCCESS, 'No VM operation required on Linux.')
```

Comment:
Can we update the response to indicate that there just is no Finch VM on Linux, instead of saying its not required? It's a little less ambiguous imo. Something like this, or similar:
```suggestion
        return format_result(STATUS_SUCCESS, 'Finch does not use a VM on Linux.')
```

(Same comment applies to other VM operations in this file, we may want to refactor this error message into the "constants" file)

Reply:
didnt move to the constant file probably need to think error messaging as a whole and move it out. that will be a todo for the future.